data_pipeline/data_processing/aggregatedcsv2redis.py

```diff
@@ -1,87 +1,179 @@
+    def setElectionShapeInRedis(self) -> list:
         '''
-        filename: str Input aggregated csv file. Csv seems to be read faster then Excel.
         Populate election shape into redis
         '''
+        data = self.data
+        electionShape = {'Elections': {}}
+        dataAmount = data[['Ballot Item', 'CandidateControlledName', 'Election Date', 'Amount', 'Rec_Type', 'ID']]
+        try:
+            # There are 4 types RCPT, EXPN, LOAN, S497.
+            elections = {}
+            for ed in dataAmount['Election Date'].unique():
+                dataPerElectionDate = dataAmount[dataAmount['Election Date'] == ed]
+                totalContributions = dataPerElectionDate[dataPerElectionDate['Rec_Type'] == 'RCPT'][
+                    'Amount'].sum().round(decimals=2)
+                officeElections = []
+                referendums = []
+                for bi in dataPerElectionDate['Ballot Item'].unique():
+                    dataPerElectionDateAndBallotItem = dataPerElectionDate[dataPerElectionDate['Ballot Item'] == bi]
+                    totalContributionsPerBallotItem = \
+                        dataPerElectionDateAndBallotItem[dataPerElectionDateAndBallotItem['Rec_Type'] == 'RCPT'][
+                            'Amount'].sum().round(decimals=2)
+                    if not 'measure' in bi:
+                        officeElections.append(
+                            {'Title': bi, 'CandidateIDs': dataPerElectionDateAndBallotItem['ID'].unique().tolist(),
+                             'TotalContributions': totalContributionsPerBallotItem})
+                    else:
+                        referendums.append(
+                            {'Title': bi, 'Description': bi, 'TotalContributions': totalContributionsPerBallotItem})
+                elections = {'Title': '{} Election Cycle'.format(ed.split('/')[2]),
+                             'Date': ed,
+                             'TotalContributions': totalContributions,
+                             'OfficeElections': officeElections,
+                             'Referendums': referendums}
+                electionShape['Elections'][ed] = elections
+                electionShape['Metadata'] = self.metadata
+        except Exception as e:
+            logger.debug(e)
+            return False
+        with self.rj.pipeline() as pipe:
+            pipe.jsonset('elections', Path.rootPath(), electionShape)
+            pipe.execute()
+        print('The election shape in redis {}'.format(self.rj.jsonget('elections')))
+        logger.debug('The election shape in redis {}'.format(self.rj.jsonget('elections')))
+        return True
+
+    def setCandidateShapeInRedis(self, electionDate='11/3/2020') -> bool:
+        '''
+        Populate candidate shape into redis
+        Redis data spec
+        Candidates: [{
+            ID: "councilmember-district-6;dev-davis;11-3-2020",
+            Name: "Dev Davis",
+            TotalRCT: 300,
```

Comment:
Do we still want to include TotalFunding as a field here?

Reply:
After today's discussion with UI, they asked for the TotalFunding.

Reply:
gotcha, is that included here? i.e. 'TotalFunding': 1000. maybe i missed it?

Reply:
Fixed typo. We can handle TotalFunding later once we figure out how LOAN count and if TotalFunding = LOAN + RCPT.